### ISSUE TITLE
StopFillCacheException is defined twice and the one is actually thrown is private and not catchable (Jira: LUCENE-4106)

### ISSUE THREAD

**Issue by mocobeta** [mocobeta/sandbox-lucene-10557#61]
```Java
lucene/core/src/java/org/apache/lucene/search/FieldCacheImpl.java
145:  static final class StopFillCacheException extends RuntimeException {
383:        } catch (StopFillCacheException stop) {
456:        } catch (StopFillCacheException stop) {
560:        } catch (StopFillCacheException stop) {
710:        } catch (StopFillCacheException stop) {
798:        } catch (StopFillCacheException stop) {
887:        } catch (StopFillCacheException stop) {

lucene/core/src/java/org/apache/lucene/search/FieldCache.java
57:  public static final class StopFillCacheException extends RuntimeException {
216:        throw new FieldCacheImpl.StopFillCacheException();
232:        throw new FieldCacheImpl.StopFillCacheException();
248:        throw new FieldCacheImpl.StopFillCacheException();
264:        throw new FieldCacheImpl.StopFillCacheException();
```

---
### Jira information

Original Jira: https://issues.apache.org/jira/browse/LUCENE-4106
Reporter: selckin
Assignee: Uwe Schindler (`@uschindler`)
Created: 2012-06-04T09:39:10.714+0000
Updated: 2016-05-09T18:31:53.697+0000
Resolved: 2012-06-17T14:07:10.053+0000

Issue Links:

Sub-Tasks:

Pull Requests:

**Comment by mocobeta** [comment]
Ah, the public declaration in the FC interface is obsolete and unusded, we should remove it.

The Exception is internal and not be intended to be used outside of Lucene's core.

Author: Uwe Schindler (`@uschindler`)
Created: 2012-06-04T09:54:27.906+0000
Updated: 2012-06-04T09:54:27.906+0000

**Comment by mocobeta** [comment]
Easy patch. This one removes the obsolete public declaration in Lucene trunk.

If yor really want to catch this internal Exception, you can do this via RuntimeException.

Author: Uwe Schindler (`@uschindler`)
Created: 2012-06-04T10:01:59.047+0000
Updated: 2012-06-04T10:01:59.047+0000

**Comment by mocobeta** [comment]
Simple patch again, this time using the public Exception, removing the private one.

I have no preference, but the duplication has to be fixed (possibly also in 3.6)

Author: Uwe Schindler (`@uschindler`)
Created: 2012-06-04T10:11:52.111+0000
Updated: 2012-06-04T10:11:52.111+0000

**Comment by mocobeta** [comment]
Committed trunk revision: 1351108
Committed 4.x revision: 1351109

Lucene 3.6.0 is not broken, it only has the private Exception.

Author: Uwe Schindler (`@uschindler`)
Created: 2012-06-17T14:07:10.170+0000
Updated: 2012-06-17T14:07:10.170+0000
